lib/miq_expression/count_field.rb

```diff
@@ -4,6 +4,10 @@ class MiqExpression::CountField < MiqExpression::Target
 \.(?<associations>[a-z_\.]+)
 /x
 
+  def self.match(field)
+    self::REGEX.match(field)
+  end
```

Comment:
Minor, but I considered moving this into the base class MiqExpression::Target, since it's identical to the one in MiqExpression::Tag, however I ended up putting them separate so we can optimize them separately.